WordPress/src/main/java/org/wordpress/android/ui/reader/viewholders/BloggingLikersTextViewHolder.kt

```diff
@@ -35,20 +35,21 @@ class BloggingLikersTextViewHolder(
 
         numBloggers.text = with(bloggersTextItem) {
             val fullText = uiHelpers.getTextOfUiString(itemView.context, textWithParams)
-            val closure = uiHelpers.getTextOfUiString(itemView.context, underlineDelimiterClosure)
-            SpannableString(fullText).formatWithSpan(itemView.context, fullText, closure)
+            formatWithSpan(itemView.context, fullText)
         }
     }
 
-    private fun SpannableString.formatWithSpan(context: Context, text: CharSequence, closure: CharSequence): Spannable {
-        val textString = StringBuffer(text).toString()
-        val closureString = StringBuffer(closure).toString()
-        val start = 0
-        val end = textString.lastIndexOf(closureString) - 1
-        return if (end <= start || end >= text.length - 1) {
-            this
-        } else {
-            this.apply {
+    private fun formatWithSpan(context: Context, text: CharSequence): Spannable {
+        val fullText = StringBuffer(text).toString()
+        val start = fullText.indexOf("_")
+        val end = fullText.lastIndexOf("_") - 1
+        val fullTextSanitized = fullText.replace("_", "")
```

Comment:
I notice that this won't work if we end up with multiple `_spans_` in the string (e.g. `Like _this_ and _that_`) as it will format the whole `this and that` instead of formatting `this` and `that` separately and keep ` and ` unchanged.

That might not be an issue with the current strings and translations this method is used on, but we don't validate what comes from translators via GlotPress so some might end up using multiple `_spans_` in their translations in some locales, who knows.

Do you think it would be too complex to update this code to instead iterate over all the `_` pairs on the string? (Either iterate on each instance of `_` found, and use a `val toggle: Bool` to track which is an opening or closing one, or use some `_[^_]*_` regular expression and iterate on all matches)? 🤔

Reply:
Hey @AliSoftware, thanks for looking into this. Yep the current implementation is making an assumption on the strings format and thinking to it more, it doesn't sound great 🙃. I like @ashiagr [suggestion](https://github.com/wordpress-mobile/WordPress-Android/pull/15329#pullrequestreview-755947219) that seems more scalable and generic so I'm going to push modifications in that direction 👍